internal/pkg/template/diff/override.go

```diff
@@ -34,18 +36,65 @@ func (m *ignorer) match(_, _ *yaml.Node, key string) bool {
 }
 
 // Parse is a no-op for an ignorer.
-func (m *ignorer) parse(_, _ *yaml.Node, _ string) (diffNode, error) {
+func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error) {
 	return nil, nil
 }
 
-type noopOverrider struct{}
+// intrinsicFuncFullShortFormConverter handles comparison between full/short form of an intrinsic function.
+type intrinsicFuncFullShortFormConverter struct{}
 
-// Match always returns false for a noopOverrider.
-func (*noopOverrider) match(_, _ *yaml.Node, _ string) bool {
-	return false
+// match returns true if from and to node represent the same intrinsic function written in different (full/short) form.
+// Example1: "!Ref abc" and "Ref: abc" will return true.
+// Example2: "!Ref abc" and "!Ref abc" will return false because they are written in the same form (i.e. short).
+// Example3: "!Ref abc" and "Fn::GetAtt: abc" will return false because they are different intrinsic functions.
+// For more on intrinsic functions and full/short forms, read https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ToJsonString.html.
+func (_ *intrinsicFuncFullShortFormConverter) match(from, to *yaml.Node, _ string, _ overrider) bool {
+	if from == nil || to == nil {
+		return false
+	}
+	if from.Kind == to.Kind || from.Kind != yaml.MappingNode && to.Kind != yaml.MappingNode {
+		// A full/short form conversion always involve at least one mapping node.
```

Comment:
unless they are the same kind of node?

Reply:
maybe i'm just misunderstanding the order of operations with the `this || that && the other` - could you add some `()` to clarify?

Reply:
i think the first condition is to make sure we ignore the case when both of them are of the same type (both short or both long), but I agree more comments will be better.